_posts/2020-03-06-introducing-mirrormaker2.md

```diff
@@ -0,0 +1,285 @@
+We'll look a bit more at configuring the `KafkaMirrorMaker2` resource later on.
+
+First, something you'll be keen to know.
+Is the sequel better than the original?
+Well, how does **_bidirectional replication_** sound?
+And what about **_topic configuration synchronization_**  and **_offset tracking_**?
+Pretty good, right?
+
+This is MirrorMaker unleashed, ready to fulfill the potential we always knew it had.
+
+### Bidirectional opportunities
+
+In the previous version of MirrorMaker, the topic name in the source cluster is automatically created in the downstream cluster.
+Fine, to a point.
+But there is no distinction between _Topic-1.Partition-1_ in the source cluster and _Topic-1.Partition-1_ in the target cluster.
+Essentially, you are limited to _active/passive_ synchronization.
+MirrorMaker 2.0 solves this by introducing the concept of _remote_ topics.
+
+_Remote_ topics are created from the originating cluster by the `MirrorSourceConnector`.
+They are distinguished by automatic renaming that appends the name of cluster to the name of the topic.
+Our _Topic-1.Partition-1_ from the source cluster becomes the never-to-be-confused _Cluster-1-Topic-1.Partition-1_ in the target cluster.
+
+As you can see here, _remote_ topics can be easily identified so that they are not returned to the source cluster.
+
+![Topic renaming with MirrorMaker 2.0](/assets/2020-03-12-mirrormaker-renaming.png)
+
+Why _remote_ topic and not just _source_ topic?
+Because each source cluster can also be a target cluster in a bidirectional configuration.
+This feature also means that it is no longer necessary to aggregate data in a separate cluster.
+
+A rose might be a rose by any other name, but when it comes to replicating topics in MirrorMaker 2.0 we can't afford such ambiguity.
+Particularly if we want to create a bidirectional configuration.
+
+The approach to topic renaming opens up a world of bidirectional opportunities.
+You can now have an _active/active_ cluster configuration that feeds data to and from each cluster.
+For this, you need a MirrorMaker 2.0 cluster at each destination.
+
+### Self-regulating topic replication
+
+Topic replication is defined using regular expression patterns to _whitelist_ or _blacklist_ topics:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker2
+spec:
+  # ...
+  mirrors:
+  - sourceCluster: "my-cluster-source"
+    targetCluster: "my-cluster-target"
+    topicsPattern: ".*"
+    # ...
+```
+
+You use `topicsBlacklistPattern` if you want to use blacklists.
+
+Pretty similar to before, but previously topic _configuration_ was not replicated.
+The target cluster had to be manually maintained to match topic configuration changes in the source cluster.
+
+This time the topic configuration is automatically synchronized between source and target clusters through configuration properties.
```

Comment:
```suggestion
With MM2 the topic configuration is automatically synchronized between source and target clusters through configuration properties.
```

But what are "configuration properties". Does it mean "this can be done by configuring it in the MM2 CR"? If that's the configuration you mean then say that.

Reply:
I changed to:

> _With MirrorMaker 2.0, the topic configuration is automatically synchronized between source and target clusters according to the topics defined in the `MirrorMaker2` custom resource._